lib/money/allocator.rb

```diff
@@ -119,7 +116,7 @@ def amounts_from_splits(allocations, splits, subunits_to_split = subunits)
       left_over = subunits_to_split
 
       amounts = splits.collect do |ratio|
-        frac = (Helpers.value_to_decimal(subunits_to_split * ratio) / allocations).floor
+        frac = (subunits_to_split * ratio / allocations.to_r).floor
```

Comment:
We actually had precision issues using decimal with very low numbers (1..10 divided by 1..10) - and changed to rational for that reason.
In this case I wonder if it would be safer to convert the dividend to rational too, the reason being:

```ruby
refute((BigDecimal(10) / 3.to_r) * 3 === 10)
assert((BigDecimal(10).to_r / 3.to_r) * 3 === 10)
```
And I'm not sure that the dividend can be bigdecimal or not :/

Reply:
From my understanding, `subunits_to_split * ratio` is supposed to always return a Rational since `subunits_to_split` is an Integer and `ratio` is a Rational. 
But I can enforce it just to be sure ! 
@elfassy : What do you think ?

Reply:
that's correct, subunits are always an integer, and integer * rational always gives a rational

Reply:
Fair point.

Reply:
@elfassy: Actually I found out that you could send all the parameters in the type that you want (for example when using `allocate`).
I think it would be a good idea to convert the parameters in the type that we want. 
What do you think ?

Reply:
Adding here our IRL conversation:
👍 good catch. Now that you mention it, this method to sum decimals also doesn't make sense
```ruby
if all_rational?(splits)
  allocations = splits.inject(0) { |sum, n| sum + n }
else
  allocations = splits.inject(0) { |sum, n| sum + Helpers.value_to_decimal(n) }
end
```
it can be replaced with with 
```ruby
splits.map!(&:to_r)
allocations = splits.inject(0, :+)
# ...
amounts, left_over = amounts_from_splits(allocations, splits)
```
This private method could raise an argument error is anyone passes a non-rational number. I think this is preferable to converting them in `amounts_from_splits` since there's always a risk in converting a float/decimal back to a rational, and this risk should probably be assumed by a public method.